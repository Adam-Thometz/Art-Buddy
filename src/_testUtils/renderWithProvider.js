import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;