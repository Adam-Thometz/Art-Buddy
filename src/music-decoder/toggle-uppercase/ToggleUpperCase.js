import React from 'react';

import { useDispatch, useSelector } from 'react-redux';
import { toggleUpperCase } from '_redux/music-decoder/wordToMusicActions';

import Toggle from '_components/toggle/Toggle';

const ToggleUpperCase = () => {
  const { isUpperCase } = useSelector(state => state.wordToMusic);
  const dispatch = useDispatch();
  
  const handleToggle = () => dispatch(toggleUpperCase());

  return <Toggle label='UPPER CASE' currToggle={isUpperCase} toggleFn={handleToggle} />;
};

export default ToggleUpperCase;