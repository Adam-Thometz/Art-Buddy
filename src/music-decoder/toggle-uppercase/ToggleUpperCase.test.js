import React from "react";

import renderWithProvider from "../../_testUtils/renderWithProvider";

import ToggleUpperCase from "./ToggleUpperCase";

describe('ToggleUpperCase component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<ToggleUpperCase />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<ToggleUpperCase />);
    expect(asFragment()).toMatchSnapshot();
  });
});