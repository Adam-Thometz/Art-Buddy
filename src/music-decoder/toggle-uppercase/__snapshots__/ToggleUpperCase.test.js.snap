// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleUpperCase component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Toggle"
  >
    <div
      class="Toggle-switch on"
    >
      <div
        class="Toggle-button"
      />
      <span>
        ON
      </span>
      <span>
        OFF
      </span>
    </div>
    <span
      class="Toggle-label"
    >
      UPPER CASE
    </span>
  </div>
</DocumentFragment>
`;
